src/utils/command.ts

```diff
@@ -43,12 +48,15 @@ export type Args =
   | ContentArgument
   | RestArgument
 
-export function parseArgs(
+export async function parseArgs(
   commandArgs: Args[] | undefined,
-  messageArgs: string[]
-): Record<string, string | number | boolean> | null {
+  message: Message
+): Promise<Record<
+  string,
+  Guild | User | Role | string | number | boolean
+> | null> {
   if (commandArgs === undefined) return null
-
+  const messageArgs = message.content.split(' ')
```

Comment:
I'm confused what is this for. Don't we already parse arguments into a raw argument part string (`argString`) and arguments array (`rawArgs`)?

Reply:
fixed it.